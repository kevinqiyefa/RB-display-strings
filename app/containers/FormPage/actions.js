/*
 *
 * HomePage actions
 *
 */
import {
  POST_STRING_REQUEST,
  POST_STRING_SUCCESS,
  POST_STRING_FAILURE,
} from './constants';

export function postStringRequest(string) {
  return {
    type: POST_STRING_REQUEST,
    string,
  };
}

export function postStringSuccess() {
  return {
    type: POST_STRING_SUCCESS,
  };
}

export function postStringFailure(err) {
  return {
    type: POST_STRING_FAILURE,
    err,
  };
}
