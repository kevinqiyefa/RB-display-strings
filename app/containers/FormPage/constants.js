/*
 *
 * FormPage constants
 *
 */
export const POST_STRING_REQUEST = 'POST_STRING_REQUEST';
export const POST_STRING_SUCCESS = 'POST_STRING_SUCCESS';
export const POST_STRING_FAILURE = 'POST_STRING_FAILURE';
