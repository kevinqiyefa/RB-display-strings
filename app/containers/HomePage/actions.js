/*
 *
 * HomePage actions
 *
 */
import {
  FETCH_STRING_REQUEST,
  FETCH_STRING_SUCCESS,
  FETCH_STRING_FAILURE,
} from './constants';

export function fetchStringRequest() {
  return {
    type: FETCH_STRING_REQUEST,
  };
}

export function fetchStringSuccess(strings) {
  return {
    type: FETCH_STRING_SUCCESS,
    strings,
  };
}

export function fetchStringFailure(err) {
  return {
    type: FETCH_STRING_FAILURE,
    err,
  };
}
