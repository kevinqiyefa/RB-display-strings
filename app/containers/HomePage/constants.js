/*
 *
 * HomePage constants
 *
 */
export const FETCH_STRING_REQUEST = 'FETCH_STRING_REQUEST';
export const FETCH_STRING_SUCCESS = 'FETCH_STRING_SUCCESS';
export const FETCH_STRING_FAILURE = 'FETCH_STRING_FAILURE';
