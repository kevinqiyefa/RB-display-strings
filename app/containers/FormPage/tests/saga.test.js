/**
 * Tests for FormPage sagas
 */

import { put, takeEvery } from 'redux-saga/effects';
import { POST_STRING_REQUEST } from '../constants';
import { postStringFailure } from '../actions';
import formPageSaga, { postStrings } from '../saga';

/* eslint-disable redux-saga/yield-effects */
describe('postStrings Saga', () => {
  const data = { string: 'test1' };
  const postStringsGenerator = postStrings(data);
  it('should post the string data', () => {
    const callDescriptor = postStringsGenerator.next().value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should call the postStringFailure action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = postStringsGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(postStringFailure(response)));
  });
});

describe('formPageSaga Saga', () => {
  const formPage = formPageSaga();

  it('should start task to watch for LOAD_REPOS action', () => {
    const takeEveryDescriptor = formPage.next().value;
    expect(takeEveryDescriptor).toEqual(
      takeEvery(POST_STRING_REQUEST, postStrings),
    );
  });
});
