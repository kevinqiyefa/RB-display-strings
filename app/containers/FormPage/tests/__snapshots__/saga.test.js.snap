// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`postStrings Saga should post the string data 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "http://localhost:3000/api/strings",
      Object {
        "body": "{\\"string\\":\\"test1\\"}",
        "headers": Object {
          "Content-Type": "application/json",
        },
        "method": "POST",
      },
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
